modules/core/src/main/java/org/apache/ignite/internal/processors/cache/binary/MetadataResponseMessage.java

```diff
@@ -16,180 +16,111 @@
  */
 package org.apache.ignite.internal.processors.cache.binary;
 
-import java.nio.ByteBuffer;
+import org.apache.ignite.internal.Order;
 import org.apache.ignite.internal.util.typedef.internal.S;
 import org.apache.ignite.plugin.extensions.communication.Message;
-import org.apache.ignite.plugin.extensions.communication.MessageReader;
-import org.apache.ignite.plugin.extensions.communication.MessageWriter;
 
 /**
  * Carries latest version of metadata to client as a response for {@link MetadataRequestMessage}.
  */
 public class MetadataResponseMessage implements Message {
-    /** */
+    /** Response status if any exception happened during preparing response. */
+    private static final byte ERROR = -1;
```

Comment:
It seems we could get rid of this status. It is never used on a receiver side.

In that case we could get rid of other statuses too and analyze just presence of `metaVerInfo`.